modules/tinymce/src/core/main/ts/keyboard/NavigationUtils.ts

```diff
@@ -80,7 +80,7 @@ const moveVertically = (editor: Editor, direction: LineWalker.VDirection, range:
   const forwards = direction === LineWalker.VDirection.Down;
 
   if (!caretClientRect) {
-    return null;
+    return Optional.none();
```

Comment:
how come our return types didn't catch the null?

Reply:
I mentioned that in the description:
> This ideally would be picked up by TS, but we'd need to turn on strict null checks which causes many TS errors.

So because TS tries to be "gentle" it allows `null` and `undefined` to match the return type by default. You have to enable strict checks to get it to fail.